agroal-pool/src/main/java/io/agroal/pool/ConnectionPool.java

```diff
@@ -448,12 +460,38 @@ public long awaitingCount() {
         return synchronizer.getQueueLength();
     }
 
+    // --- health check //
+
+    @Override
+    public boolean isHealthy(boolean connect) throws SQLException {
+        ConnectionHandler healthHandler;
```

Comment:
If the pool is full, and all connections are checked out - this needs to work as well